src/Agent.Plugins/PipelineCache/TarUtils.cs

```diff
@@ -0,0 +1,232 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.ExceptionServices;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Agent.Sdk;
+using Microsoft.VisualStudio.Services.BlobStore.Common;
+using Microsoft.VisualStudio.Services.BlobStore.WebApi;
+
+namespace Agent.Plugins.PipelineCache
+{
+    public static class TarUtils
+    {
+        private readonly static bool isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+
+        private const string archiveFileName = "archive.tar";
+
+        /// <summary>
+        /// Will archive files in the input path into a TAR file.
+        /// </summary>
+        /// <returns>The path to the TAR.</returns>
+        public static async Task<string> ArchiveFilesToTarAsync(
+            AgentTaskPluginExecutionContext context,
+            string inputPath,
+            CancellationToken cancellationToken)
+        {
+            var archiveFile = Path.Combine(Path.GetTempPath(), archiveFileName);
```

Comment:
there should be a guid so multiple agents don't collide

Reply:
I did have a GUID initially, but after discussing with @owenhuynMSFT decided to remove it. And as the agent tasks run in sequence we shouldn't have collision.

Reply:
Plus, we needed for manifest validation.